modules/ROOT/pages/using-flows-for-service-orchestration.adoc

```diff
@@ -0,0 +1,128 @@
+= Flows
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: anypoint studio, studio, mule esb, orchestration
+
+== Introduction
+
+A *flow* is a simple yet very flexible mechanism that enables orchestration of services using the sophisticated message flow capabilities of Mule. Using a flow, you may automate integration processes and construct Mule message processing solutions by ordering any number of elements of a Mule flow in a valid arrangement. Because what you do inside a flow is up to you as the developer, it is much easier to create solutions that match your requirements.
+
+== When to Use a Flow
+
+[TIP]
+*A flow is the most versatile and powerful integration mechanism available in Mule.*
+
+Flows are valuable in many situations, including:
+
+* Simple integration tasks
+* Scheduled data processing
+* Connecting cloud and on-premise applications
+* Event processing where multiple services need to be composed
+
+
+== The Anatomy of a Flow
+
+A flow is in essence just a chain of Mule Components. Think of each component as a Lego block where a Flow is something you build with them. A flow also has a message source, the source of messages that are processed by the component chain.
+
+//image::flow.jpg[]
+
+== Flow Configuration
+
+A Flow is configured in XML using the <flow> element. Each flow has a name attribute, a message source (unless it's a private flow), one or more components and an optional error handler.
```

Comment:
Use the XML <flow> element to configure a flow (or this text: Use the flow element in XML to configure a flow.)

Remove the "a" before "message source" and add a comma after "components"